lib/simctl.js

```diff
@@ -75,7 +75,11 @@ async function createDevice (name:string, deviceTypeId:string,
 }
 
 async function deleteDevice (udid:string):void {
-  await simExec('delete', 0, [udid]);
+  let loopFn:Function = async () => {
+    await simExec('delete', 0, [udid]);
+  };
+  // retry delete with a sleep in between because it's flakey
+  await retryInterval(5, 200, loopFn);
```

Comment:
Yeah. I'm fixing that. 🤔

On Thursday, November 19, 2015, Jonah <notifications@github.com> wrote:

> In lib/simctl.js
> <https://github.com/appium/node-simctl/pull/13#discussion_r45379951>:
>
> > @@ -75,7 +75,11 @@ async function createDevice (name:string, deviceTypeId:string,
> >  }
> >
> >  async function deleteDevice (udid:string):void {
> > -  await simExec('delete', 0, [udid]);
> > +  let loopFn:Function = async () => {
> > +    await simExec('delete', 0, [udid]);
> > +  };
> > +  // retry delete with a sleep in between because it's flakey
> > +  await retryInterval(5, 200, loopFn);
>
> Ah yeah. The create function is the one that needs to be fixed. when you
> call the create function it returns right away and if you run simctl list
> it shows up in the Creating state. If we wait a little bit it should go
> into Shutdown state, at which point we can delete it. The real fix here
> should be changing the create function to start polling sim.stat() until
> the state is 'Shutdown` and then return.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appium/node-simctl/pull/13/files#r45379951>.
>